### ISSUE TITLE
Classes in mustache.js 2.0.0 ?

### ISSUE THREAD

**Issue by leonardo-silva** [janl/mustache.js#446]
Hello,

As part of my PhD research, I am working on an approach to identify potential 
classes in JavaScript software (ES 5). More details can be found in the paper
Conference (http://www.dcc.ufmg.br/~mtov/pub/2015_saner.pdf).

I have applied my approach to version 2.0.0 of mustache and I found 3 classes (see attached UML diagram)

It would be really important to my research if you could answer a couple of questions:

- Do you agree with these findings (for the classes, methods and attributes)? 

- Do you intend to use the new support for classes that comes with ECMAScript 6 ? Why ?

Thank you in advance!!

Leonardo Humberto Silva
Federal University of Minas Gerais - Brazil

![classesformustache2-0-0](https://cloud.githubusercontent.com/assets/7905740/7390552/aef51eee-ee79-11e4-9bb4-48571a63db77.png)

**Comment by phillipj** [comment]
- Do you agree with these findings (for the classes, methods and attributes)?

Yes.

Out of curiosity; what are you planning to call object literals with methods attached to them in your paper? E.g. such as our main API the `mustache` object. There's a lot of them around in the world of javascript.

- Do you intend to use the new support for classes that comes with ECMAScript 6 ? Why ?

I dont remember any discussions about using ES6 so far. In other words its not planned to do so very soon. At the moment we dont transpile or pre-process our code, which means we're left waiting for the majority of browsers to have ES6 classes.

All that might change in the future of course, I cant predict we're up to in the distant future.

**Comment by leonardo-silva** [comment]
Thank you for the answers!
Good question on the object literals. When I tried to classify them as classes I got too many false positives. They are used for many different reasons inside js applications. In most of the cases I analyzed, they were not intended to be classes. And, the idea of my approach is to be able to identify the classes without any knowledge of the system (without any manual inspection). I just pass the path where the source code is located and the tool draws the uml diagram.
That is the main reason why I do not classify object literals as classes. One idea, though, is to give the user an option to see all object literals that the system has. That is perfectly possible.
Best,
